pkg/securityscan/controller.go

```diff
@@ -153,3 +165,49 @@ func detectKubernetesVersion(ctx context.Context, k8sClient kubernetes.Interface
 	}
 	return v.GitVersion, nil
 }
+
+func initializeMetrics(ctl *Controller) error {
+	ctl.numTestsFailed = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "cis_scan_num_tests_fail",
+			Help: "Number of test failed in the CIS scans, partioned by scan_name, scan_profile_name",
+		},
+		[]string{
+			// scan_name will be set to "manual" for on-demand manual scans and the actual name set for the scheduled scans
+			"scan_name",
+			// name of the clusterScanProfile used for scanning
+			"scan_profile_name",
+			"num_tests_total",
+			"num_tests_skip",
+			"num_tests_na",
+		},
+	)
+
+	if err := prometheus.Register(ctl.numTestsFailed); err != nil {
+		return err
+	}
+
+	ctl.numScansComplete = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "cis_scan_num_scans_complete",
+			Help: "Number of CIS clusterscans completed, partioned by scan_name, scan_profile_name",
+		},
+		[]string{
+			// scan_name will be set to "manual" for on-demand manual scans and the actual name set for the scheduled scans
+			"scan_name",
+			// name of the clusterScanProfile used for scanning
+			"scan_profile_name",
+			"num_tests_total",
+			"num_tests_skip",
+			"num_tests_na",
+			"num_tests_pass",
+			"num_tests_fail",
+		},
+	)
```

Comment:
For `num_tests_total`, `num_tests_skip`, and `num_tests_na`, can we assume that these are unlikely to change frequently for a given `scan_profile_name` (e.g. is it okay to store 1 series per combination of those labels)? I know that [`num_tests_skip` seems to be constant for a given ScanProfile](https://github.com/rancher/cis-operator/blob/0e3ca2f26a7ff33498d17c32c3edcdd293bc282d/pkg/apis/cis.cattle.io/v1/types.go#L123) but I'm not sure if we can assume the same for the other two.

Reply:
However, I think that `num_tests_pass` and `num_tests_fail` definitely need to be removed from `cis_scan_num_scans_complete` since those are determined on a per-scan basis and will likely balloon the number of series required. Is that information necessary for the `alertOnComplete` alert or could that be safely removed?

Reply:
yes most of this infomation is required to be in the alert content sent. I dont see any other way to set this content without using labels. `num_tests_total`, `num_tests_skip` and `num_tests_na` will remain constant for a given profile. We could include either of the  `num_tests_fail` or `num_tests_pass` and derive the other - that will eliminate one label.